addons/project_timesheet_holidays/models/res_company.py

```diff
@@ -1,50 +1,44 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models, _
-from odoo.exceptions import ValidationError
+from odoo import fields, models, _
 
 
 class Company(models.Model):
     _inherit = 'res.company'
 
-    leave_timesheet_project_id = fields.Many2one(
-        'project.project', string="Internal Project",
-        help="Default project value for timesheet generated from time off type.")
     leave_timesheet_task_id = fields.Many2one(
         'project.task', string="Time Off Task",
-        domain="[('project_id', '=', leave_timesheet_project_id)]")
-
-    @api.constrains('leave_timesheet_project_id')
-    def _check_leave_timesheet_project_id_company(self):
-        for company in self:
-            if company.leave_timesheet_project_id:
-                if company.leave_timesheet_project_id.sudo().company_id != company:
-                    raise ValidationError(_('The Internal Project of a company should be in that company.'))
+        domain="[('project_id', '=', internal_project_id)]")
 
     def init(self):
         type_ids = [(4, self.env.ref('hr_timesheet.internal_project_default_stage').id)]
-        for company in self.search([('leave_timesheet_project_id', '=', False)]):
+        companies = self.search(['|', ('internal_project_id', '=', False), ('leave_timesheet_task_id', '=', False)])
+        internal_projects_by_company_dict = None
+        Project = self.env['project.project']
+        for company in companies:
             company = company.with_company(company)
-            project = company.env['project.project'].search([
-                ('name', '=', _('Internal')),
-                ('allow_timesheets', '=', True),
-                ('company_id', '=', company.id),
-            ], limit=1)
-            if not project:
-                project = company.env['project.project'].create({
-                    'name': _('Internal'),
-                    'allow_timesheets': True,
-                    'company_id': company.id,
-                    'type_ids': type_ids,
-                })
-            company.write({
-                'leave_timesheet_project_id': project.id,
-            })
+            if not company.internal_project_id:
+                if not internal_projects_by_company_dict:
+                    internal_projects_by_company_read = Project.search_read([
+                        ('name', '=', _('Internal')),
+                        ('allow_timesheets', '=', True),
+                        ('company_id', 'in', companies.ids),
+                    ], ['company_id', 'id'])
+                    internal_project_by_company_dict = {res['company_id'][0]: res['id'] for res in internal_projects_by_company_read}
```

Comment:
isn't there a typo here? Shouldn't it be internal_project**s**_by_company_dict?

Reply:
@IT-Ideas , Done the changes. Thanks